### PR TITLE
Generate a history profile given a records history.

### PR DESCRIPTION
**Pull request by jannistsiroyannis**
The profile will tell you "who" is responsible for each piece
of information in the record. The idea is to use this to decide
if parts of incoming updates should be merged/added/discarded.

### CODE REVIEW

File: whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy [hunk 1]
```diff
@@ -2290,6 +2292,29 @@ class PostgreSQLComponent {
...
+            ResultSet rs = null
+            List<DocumentVersion> docList = []
+            try {
+                selectstmt = connection.prepareStatement(GET_ALL_DOCUMENT_VERSIONS)
+                selectstmt.setString(1, id)
+                rs = selectstmt.executeQuery()
+                int v = 0
+                while (rs.next()) {
+                    def doc = assembleDocument(rs)
+                    doc.version = v++
```
Comment (olovy) [inline whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy:2308]: Never used as far as I can tell, but it numbering is backwards?
v 0 == current
Or is that intentional?

File: whelk-core/src/main/groovy/whelk/history/Ownership.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import java.time.ZonedDateTime;
+
+public class Ownership {
+    public String m_lastManualEditor;
+    public Instant m_lastManualEditTime;
+
+    public String m_lastEditor;
+    public Instant m_lastEditTime;
+
+    public Ownership(DocumentVersion version, Ownership previousOwnership) {
```
Comment (olovy) [inline whelk-core/src/main/groovy/whelk/history/Ownership.java:13]: Libris-specific stuff, but I think that is ok for now. Let's just note increased "Libris debt" in whelk-core.
(We could later e.g. move interpretation of the strings xl/Voyager/vcopy/APIX etc. to importers?)

**Review by olovy — APPROVED** [review]
LGTM.

A good starting point.
Let's add some tests later.

File: whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy (hunk 1 above)
Comment (olovy) [reply whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy:2308]: Ok, same numbering as in `doLoadAllVersions`

Comment (jannistsiroyannis) [reply whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy:2308]: > Ok, same numbering as in `doLoadAllVersions`

Yeah, its used here (where that needs to be the order):
java/whelk/export/marc/ProfileExport.java:174

**Pull request merged**